### DISCUSSION TITLE
avoid '__state__completed' jobs?

### DISCUSSION THREAD

**Discussion by tommygoessing** [timgit/pg-boss#229] [Q&A] [answered]
hi,

our implementation removes completed jobs from the jobs table and creates a new succeeding job with the same name, prefixed by "__state_completed" for each one of them.

Is this a configurable behaviour? or fixed and non manipulatable behaviour? (couldn't find the answer from the documentation)

If fixed: Can you quickly explain why?

We don't see a real sense in those new" __state_completed..." jobs, as they don't hold any more information, then the original job in state "completed"can, on his own...

best,
Tom

**Comment by brianmcd** [comment, marked as answer]
Those jobs are how pg-boss implements the `onComplete` functionality. More info on onComplete: https://github.com/timgit/pg-boss/blob/master/docs/usage.md#oncompletename--options-handler.

As of version 5.2.0, you can control the creation of the `__state_completed` jobs (either per-publish or globally). https://github.com/timgit/pg-boss/blob/master/docs/configuration.md#completion-jobs
